src/components/ApplicationMessages/index.js

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect, useState } from "react";
+import PropTypes from "prop-types";
+import { addSnackbar } from "../../actions/notificationActions";
+import useSession from "../../hooks/useSession";
+import { connect } from "react-redux";
+import Notification from "../Notifications/Notification";
+import { useHistory } from "react-router";
+
+const ApplicationMessageNotifier = ({ addSnackbar }) => {
+
+    const [hasShownCompleteRegistrationMessage, setHasShownCompleteRegistrationMessage] = useState(false);
+    const [hasRevalidated, setHasRevalidated] = useState(false);
+
+    const history = useHistory();
+
+    const { data, isLoggedIn, revalidate, isValidating } = useSession();
+
+    useEffect(() => {
+        if (!hasRevalidated || isValidating) return;
+        if (isLoggedIn && !data.isAdmin && !data.company?.hasFinishedRegistration && !hasShownCompleteRegistrationMessage) {
+            setHasShownCompleteRegistrationMessage(true);
+            addSnackbar({
+                message: "In order to fully use NIJobs, you still need to finish your registration.",
+                options: {
+                    // eslint-disable-next-line react/display-name
+                    content: (key, message) =>
+                        <Notification
+                            message={message}
+                            actionText="Show me"
+                            actionHandler={() => {
+                                history.push("/company/registration/finish");
+                            }}
+                        />,
+                },
+            });
+        }
+        if (hasShownCompleteRegistrationMessage && !isLoggedIn) setHasShownCompleteRegistrationMessage(false);
```

Comment:
What is this line for? I imagine it is useful when the notification has been shown and the user logs out, in which case it is not supposed to show the notification on page "reload", please correct me if I'm wrong.

Reply:
It's to prevent the message from showing multiple times on component remount (not necessarily page reload). Without it, the notification would show every time we go to another page inside this website, since it does not reload (it's a SPA), it simply remounts another subtree of components (which would make this component re-render and show the message again)

This component will also re-render on logout, but the if above would check `isLoggedIn` in order to show the notification. However, if the user would log back in, we want the notification to show again, which would not happen since the `hasShownCompleteRegistrationMessage` is true, so it works kind of like a reset.

Nevertheless, the usage of `hasShownCompleteRegistrationMessage` is important to keep the notification from showing on every re-render (while the user is logged in and without a complete registration)